You-Can-Code-It/fndr
# Delete company and add company to list

# What this pull request does

## What

- A company can be removed from the list of displayed ones
- Also, a new company can be added to the list.

## Why

- It is pointless to show users outdated data, for example, a company that is no longer hiring
- On the other hand, the list of companies shall have the possibility to grow, displaying new companies or companies that were not originally added to the database

## How

- We're choosing not to delete a company from the database, but not to display it:
- To do that, we're adding a new field in the Company model: display (boolean true as default)
- When the admin confirms the removal (using a modal), an update request is sent to the database: "display= false"
- When getting the props in the main page, the array of company objects is filtered to display only the companies whose "display= true"
- To add a new company, another component "AddCompanyForm" is created, and opens when the user clicks the "Add" button in the main page
- This form uses `React-hook-form` combined with `Zod validation` so the admin gets a feedback in case some field is not accordingly filled
- The validation uses `Regex` rules for checking the "website" and "zip code" fields
- An api POST request is made to insert the data to the Company table

## Steppings to test this

(a) Remove feature: 
- Go to the main page, choose a company and click the "Remove" button: a modal with a message must appear, asking to confirm or cancel
- Click "Confirm" and the company card should be no longer appearing (check also the companies' counter on the top left side of the page, that will automatically show one unit less than before)

(b) Add new company feature:

- Go to the main page, and click the "Add" button
- A form should appear
- Try to submit with a missing field or for example typing an invalid Dutch zip code (something different than four numbers followed by two letters). In this case, a correspondent error message will appear
- Submit the form (with correct input), and check if a new entry was added to the Company's table

## Anything else

- I started with the remove feature, than came to the conclusion that the add new company one could be also implemented in the same branch, due to the existing similarities between them, but it could have been done in another branch for better scope division purposes
